test/integration/metrics_cgroupsv2_test.go

```diff
@@ -0,0 +1,314 @@
+package integration_test
+
+import (
+	"github.com/newrelic/infra-integrations-sdk/persist"
+	"github.com/newrelic/nri-docker/src/biz"
+	"github.com/newrelic/nri-docker/src/raw"
+	"github.com/newrelic/nri-docker/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+)
+
+const (
+	cgroupDriver                              = "systemd"
+	InspectorPIDCgroupsV2                     = 667
+	relativePathToTestdataFilesystemCgroupsV2 = "testdata/cgroupsV2_host/"
+)
+
+func TestCgroupsV2HighCPU(t *testing.T) {
+	containerID, dockerRM := stress(t, "stress-ng", "-c", "4", "-l 100", "-t", "5m")
+	docker := newDocker(t)
+
+	t.Cleanup(func() {
+		dockerRM()
+		docker.Close()
+	})
+
+	cgroupFetcher, err := raw.NewCgroupsV2Fetcher("/", raw.NewPosixSystemCPUReader(), cgroupDriver)
+	require.NoError(t, err)
+
+	metrics := biz.NewProcessor(
+		persist.NewInMemoryStore(),
+		cgroupFetcher,
+		docker,
+		0)
+
+	sample, err := metrics.Process(containerID)
+	require.NoError(t, err)
+
+	assert.InDelta(t, cpus, sample.CPU.LimitCores, 0.01)
+	assert.True(t, sample.Pids.Current >= 4, "pids need to be >= 4") // because of stress-ng -c 4
+
+	test.Eventually(t, eventuallyTimeout, func(t require.TestingT) {
+		time.Sleep(100 * time.Millisecond)
+		sample, err = metrics.Process(containerID)
+		require.NoError(t, err)
+
+		cpu := sample.CPU
+
+		// the samples tend to show CPU metrics near to their limits
+		//assert.InDelta(t, 100*cpus, cpu.CPUPercent, 15)
+		assert.InDelta(t, 100, cpu.UsedCoresPercent, 15)
+		assert.InDelta(t, cpus, cpu.UsedCores, 0.2)
+
+		//assert.True(t, cpu.UserPercent > 0,
+		//	"user percent not > 0")
+		//assert.True(t, cpu.KernelPercent >= 0,
+		//	"kernel percent not >= 0")
+
+		//assert.Truef(t, cpu.UserPercent+cpu.KernelPercent <= cpu.CPUPercent,
+		//	"user %v%% + kernel %v%% is not < total %v%%",
+		//	cpu.UserPercent, cpu.KernelPercent, cpu.CPUPercent)
```

Comment:
This needs to be uncommented when we have the `PercpuUsage` value in cgroupsV2. Implemented in another PR by @sigilioso .

Reply:
#105 this is it. As soon as we get it merged we can update this one and uncomment.

Reply:
I miss [cpuset files](https://www.kernel.org/doc/html/latest/admin-guide/cgroup-v2.html#cpuset-interface-files) in the testdata files for cgroups2. We need, at least, `cpuset.cpus.effective` because `OnlineCPUs` value is got from there.

Reply:
Yes, I just added what used so far. In fact, when we add the new cgroupsV2 metrics, probably we'll need to add some additional files. Thanks for pointing out 👍